File: model/src/main/java/io/mateu/erp/model/world/Country.java
package io.mateu.erp.model.world;

import io.mateu.erp.model.taxes.VAT;
import io.mateu.mdd.core.annotations.Ignored;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.ArrayList;
import java.util.List;

/**
 * holder for countries
 *
 * Created by miguel on 13/9/16.
 */
@Entity
@Getter@Setter
public class Country {

    @Id
    @NotNull
    private String isoCode;

    @NotNull
    private String name;

    @OneToMany(mappedBy = "country")
    @Ignored
    private List<Destination> destinations = new ArrayList<>();

    @ManyToOne
    private VAT vat;

    @ElementCollection
    private List<String> nationalAirportsIATACodes;

    @Column(name = "_order")
    private int order;


    @Override
    public String toString() {
        return getName();
    }


    @Override
    public int hashCode() {
        return getClass().hashCode();
    }

    @Override
    public boolean equals(Object obj) {
        return this == obj || (obj != null && obj instanceof Country && isoCode == ((Country)obj).isoCode);
    }

}

File: model/src/main/java/io/mateu/erp/model/world/Country.kt
package io.mateu.erp.model.world

import io.mateu.ui.mdd.server.annotations.Ignored
import lombok.Getter
import lombok.Setter

import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.OneToMany
import javax.validation.constraints.NotNull
import java.util.ArrayList

/**
 * holder for countries
 *
 * Created by miguel on 13/9/16.
 */
@Entity
//@QLForCombo(ql = "select x.isoCode, x.name from io.mateu.erp.model.world.Country x order by x.name")
class Country {

    @Id
    @NotNull
    var isoCode: String? = null

    @NotNull
    var name: String? = null

    @OneToMany(mappedBy = "country")
    @Ignored
    var states = ArrayList<State>()
}
